addon/services/stripe.js

```diff
@@ -63,6 +64,7 @@ export default Service.extend({
     this._alive = {};
     this._scriptLoaded = false;
     this._scriptLoading = false;
+    this.stripeConfig = getOwner(this).lookup('config:stripe');
```

Comment:
@vladucu Without that, the configs that come from the implementing platform (admin) do not load. Without that key data like the stripe key set on smile-admin `confing/environment` does not load for this addon. Also, it is used [here](https://github.com/smile-io/ember-cli-stripe/pull/194/files/7f131b480dfbc39ae09486a8739a852b4e1ac1ff#diff-6339955f862dc343ebd4415d5549cde4d9ed2553b314297dcb9e7b5caba21799L81)

Reply:
> Without that, the configs that come from the implementing platform (admin) do not load.

Can you help me understand why?

Reply:
The idea behind this change was inspired by [this change](https://github.com/smile-io/ember-smile-core/commit/d759a01401c33aab0ecb31b4f5da921995bd9117) yours. Here is the [deprecation doc](https://deprecations.emberjs.com/v3.x/#toc_implicit-injections) I based on:

> [...] In cases where it is not possible to convert a custom injection type into a service, the value can be accessed by looking it up directly on the container instead using the [lookup](https://api.emberjs.com/ember/3.22/classes/ApplicationInstance/methods/lookup?anchor=lookup) method.

From what I understand now, `getOwner(this).lookup` is a general way of accessing registered objects within an Ember application, allowing you to retrieve not just services but any type of object registered with the application's container, which is the case for this addon

Reply:
Got it. Thanks!
If this is the case, let's please also remove this [line of code](https://github.com/smile-io/ember-cli-stripe/blob/main/app/initializers/stripe.js#L8) that might be confusing in the future.

PS Sorry for all the extra questions. I didn't realise that this was related to _implicit injection deprecation_ from Ember 4 because there was no change in here yo outline that. In general, it's helpful to provide answers to the _WHY (ex: we need this because in Ember 4 implicit injections are no longer supported)_ instead of _WHAT (ex: this no longer works - one can tell from reading the PR considering the change was intentional)_

Reply:
Ah, thanks @vladucu ! I already removed the redundant code ([ref](https://github.com/smile-io/ember-cli-stripe/pull/194/commits/e8be2e579989a2888f3b72bbfd934a1ac73a6cc6)). Also, I will try to keep the why/what advice in mind for next time